contracts/staking/StakedSYN.sol

```diff
@@ -0,0 +1,116 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.6.12;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+
+interface IMinter {
+    function stakingMint(uint256 lastMint) external returns(uint256);
+}
+
+contract StakedSYN is Ownable, ERC20("Staked Synapse", "sSYN") {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+    IERC20 public synapse;
+    IMinter public STAKING_MINTER;
+    // Tracks actively staked sSYN. This amounts to totalSupply() - requested unstake total
+    uint256 internal activeStaked;
+    // Tracks active delegated SYN backing sSYN
+    uint256 internal activeSYN;
+
+    // Last timestamp of SYN distributed to contract
+    uint256 internal lastSynMint;
+    
+    // Tracks when SYN can be unstaked, along with amounts at time of launch
+    mapping(address => uint256) public undelegateTimestamps;
+    mapping(address => uint256) public undelegateUnderlyingAmounts;
+
+    // Define the Synapse token contract
+    constructor(IERC20 _synapse, uint256 _stakingStart) public {
+        synapse = _synapse;
+        lastSynMint = _stakingStart;
+    } 
+
+    /*** RESTRICTED FUNCTIONS ***/
+
+    function setStakingMinter(address _minter) external onlyOwner {
+        STAKING_MINTER = IMinter(_minter);
+    }
+
+    /*** VIEW FUNCTIONS ***/
+
+    function undelegatedSynapse(address user) external view returns(uint256 amount, uint256 timestamp) {
+        return (undelegateUnderlyingAmounts[user], undelegateTimestamps[user]);
+    }
+
+    function underlyingBalanceOf(address user) external view returns(uint256) {
```

Comment:
Consider introducing an additional internal view function for that calculation, as it's used a few times in the code (lines 54, 80, 96).

```solidity
function _getUnderlyingSynapseAmount(uint256 _amount) internal view returns(uint256) {
  uint256 totalStaked = totalSupply();
  return totalStaked > 0 ? _amount.mul(activeSYN).div(totalStaked) : 0
}

function underlyingBalanceOf(address user) external view returns(uint256) {
  return _getUnderlyingSynapseAmount(balanceOf(user)) + 
         undelegateUnderlyingAmounts[user];
}
```

1. totalSupply can be zero (before the staking and when everyone undelegated).
2. User can undelegate fraction of their sSYN, so you need to track both sSYN balance and undelegated amount.